ts/nni_manager/common/training_service_v3.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to TrialJobMetric.data
+ **/
+export type Metric = string;
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to HyperParameters.value
+ **/
+export type Parameter = string;
+
+export interface EnvironmentInfo {
+    id: string;
+}
+
+export interface TrainingServiceV3 {
+    init(): Promise<void>;
+    start(): Promise<EnvironmentInfo[]>;
+    stop(): Promise<void>;
+
+    uploadDirectory(directoryName: string, path: string): Promise<void>;
+    createTrial(environmentId: string, trialCommand: string, directoryName: string): Promise<string | null>;
+    stopTrial(trialId: string): Promise<void>;
+    sendParameter(trialId: string, parameter: Parameter): Promise<void>;
+
+    onRequestParameter(callback: (trialId: string) => Promise<void>): void;
```

Comment:
One trial can still have multiple parameter?

Reply:
Yes.
Each `get_next_parameter()` results in one `onRequestParameter()`, and then one `sendParameter()`.